gensim/models/keyedvectors.py

```diff
@@ -336,6 +346,106 @@ def most_similar(self, positive=[], negative=[], topn=10, restrict_vocab=None, i
         result = [(self.index2word[sim], float(dists[sim])) for sim in best if sim not in all_words]
         return result[:topn]
 
+    def most_similar_among(self, positive=[], negative=[], topn=10, words_list=None, indexer=None,
+                            suppress_warnings=False):
```

Comment:
Actually, calculating difference in sets (which is needed to generate this warning) can be an unnecessary overhead for people using this code in a production environment. I have added a `logging.info` call to inform users of the same.
Sometimes it is not possible to remove all non-vocabulary words from the `words-list`, for example, if `words-list` is generated during runtime. Here using `suppress_warnings` would be ideal.